### PR TITLE
Verticalspace Regex bug

### PR DESCRIPTION
**Pull request by masters3d**
closes https://github.com/realm/SwiftLint/issues/772

### CODE REVIEW

**Comment by masters3d** [comment]
I left it as opt in until after review.

File: Source/SwiftLintFramework/Rules/VerticalWhitespaceRule.swift [hunk 1]
```diff
@@ -93,22 +87,16 @@ public struct VerticalWhitespaceRule: CorrectableRule,
...
-            // filtering out violations within a multiple comment block
-            let isSectionInComment = !comments.filter {
-                (eachRange, _ ) in eachRange.intersectsRange(lastLine.range)
-            }.isEmpty
-
-            if isSectionInComment {
-                continue  // skipping the lines found in multiline comment
-            } else {
+            let kindInSection = syntaxMap.tokensIn(lastLine.byteRange)
+                                                        .map { $0.type }
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/VerticalWhitespaceRule.swift]: instead of mapping twice, how about `.flatMap { SyntaxKind(rawValue: $0.type) }`?

**Comment by jpsim** [comment]
What's the before/after duration of `IntegrationTests.testSwiftLintLints()` with this change?

Also, could you please add a changelog entry? I think it's safe to remove `OptInRule` if the perf issues are resolved.

**Comment by jpsim** [comment]
Thanks for doing this!

**Comment by masters3d** [comment]
> What's the before/after duration of IntegrationTests.testSwiftLintLints() with this change?

These are just one pass. We probably need to do averages but they are about the same. I believe the issue was just a regex bug. 

```
***after ***  
Test Case '-[SwiftLintFrameworkTests.RulesTests testVerticalWhitespace]' passed (0.052 seconds).

Test Case '-[SwiftLintFrameworkTests.IntegrationTests testSwiftLintAutoCorrects]' passed (1.271 seconds).
Test Case '-[SwiftLintFrameworkTests.IntegrationTests testSwiftLintLints]' started.
Test Case '-[SwiftLintFrameworkTests.IntegrationTests testSwiftLintLints]' passed (0.839 seconds).
Test Suite 'IntegrationTests' passed at 2016-08-25 12:48:37.585.

***before****
Test Case '-[SwiftLintFrameworkTests.RulesTests testVerticalWhitespace]' passed (0.044 seconds).

Test Case '-[SwiftLintFrameworkTests.IntegrationTests testSwiftLintAutoCorrects]' passed (1.176 seconds).
Test Case '-[SwiftLintFrameworkTests.IntegrationTests testSwiftLintLints]' started.
Test Case '-[SwiftLintFrameworkTests.IntegrationTests testSwiftLintLints]' passed (0.858 seconds).
Test Suite 'IntegrationTests' passed at 2016-08-25 12:46:19.384.

```

**Comment by jpsim** [comment]
Oh, interesting... Maybe the file shared in #772 was just specially crafted to hit the pathological case with the previous regex. I hope I'm not too demanding, but could you please share the before/after with the sample code shared in #772?

**Comment by masters3d** [comment]
BEFORE
```
0.000: file_length
0.000: leading_whitespace
0.000: custom_rules
0.000: force_try
0.000: todo
0.000: force_cast
0.000: closing_brace
0.000: statement_position
0.000: legacy_constant
0.000: legacy_constructor
0.000: trailing_semicolon
0.000: trailing_whitespace
0.000: return_arrow_whitespace
0.000: legacy_cggeometry_functions
0.000: type_body_length
0.000: private_unit_test
0.000: trailing_newline
0.000: legacy_nsgeometry_functions
0.000: operator_whitespace
0.000: mark
0.000: function_body_length
0.000: control_statement
0.000: nesting
0.000: line_length
0.000: opening_brace
0.000: variable_name
0.000: type_name
0.000: valid_docs
0.000: colon
0.000: comma
0.001: cyclomatic_complexity
0.001: function_parameter_count
58.832: vertical_whitespace
```

AFTER

```
0.000: file_length
0.000: leading_whitespace
0.000: custom_rules
0.000: force_try
0.000: todo
0.000: force_cast
0.000: closing_brace
0.000: legacy_constant
0.000: statement_position
0.000: legacy_constructor
0.000: trailing_semicolon
0.000: trailing_whitespace
0.000: return_arrow_whitespace
0.000: legacy_cggeometry_functions
0.000: private_unit_test
0.000: trailing_newline
0.000: type_body_length
0.000: operator_whitespace
0.000: legacy_nsgeometry_functions
0.000: function_parameter_count
0.000: variable_name
0.000: mark
0.000: control_statement
0.000: function_body_length
0.000: nesting
0.000: line_length
0.000: opening_brace
0.000: type_name
0.000: cyclomatic_complexity
0.000: valid_docs
0.000: colon
0.000: comma
0.002: vertical_whitespace
```

**Comment by masters3d** [comment]
So this line is really slow
```swift
        let syntaxMap = SyntaxMap(file: file)

```
0.013: vertical_whitespace vs 0.485: vertical_whitespace

So now I do have a performance issue. I can avoid it by just not worrying about vertical white space inside multiline comments

```
‎/*



*/	
```

**Comment by masters3d** [comment]
okay. fixed by swapping out that line :)

```swift
        let syntaxMap = file.syntaxMap

```
benchmark on Swiftlint dir

```
0.000: custom_rules
0.000: file_length
0.000: leading_whitespace
0.002: trailing_newline
0.003: trailing_semicolon
0.003: force_cast
0.003: closing_brace
0.004: statement_position
0.004: empty_count
0.004: todo
0.004: force_try
0.006: return_arrow_whitespace
0.006: legacy_constant
0.008: operator_whitespace
0.010: legacy_constructor
0.011: line_length
0.013: legacy_nsgeometry_functions
0.013: legacy_cggeometry_functions
0.013: trailing_whitespace
0.014: private_unit_test
0.014: control_statement
0.016: vertical_whitespace
0.021: type_body_length
0.022: type_name
0.027: function_parameter_count
0.031: mark
0.033: variable_name
0.038: nesting
0.040: comma
0.040: opening_brace
0.044: colon
0.054: function_body_length
0.058: cyclomatic_complexity
0.091: valid_docs

```

**Comment by jpsim** [comment]
Thanks for this @masters3d! You forgot to add that the rule is no longer opt-in to the changelog, but it'll be faster if I fix that up in a followup commit.

**Comment by masters3d** [comment]
Thank you

**Pull request merged**